components/ai_chat/resources/page/components/open_external_link_modal/index.tsx

```diff
@@ -0,0 +1,121 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import Button from '@brave/leo/react/button'
+import Checkbox from '@brave/leo/react/checkbox'
+import Dialog from '@brave/leo/react/dialog'
+import { getLocale } from '$web-common/locale'
+import styles from './style.module.scss'
+import getAPI from '../../api'
+import {
+  IGNORE_EXTERNAL_LINK_WARNING_KEY //
+} from '../../../common/constants'
+
+export default function OpenExternalLinkModal() {
+  // API
+  const api = getAPI()
+
+  // State
+  const [ignoreChecked, setIgnoreChecked] = React.useState(false)
+  const [openingExternalLinkURL, setOpeningExternalLinkURL] = React.useState('')
+
+  // Local storage
+  const ignoreExternalLinkWarning = JSON.parse(
+    localStorage.getItem(IGNORE_EXTERNAL_LINK_WARNING_KEY) ?? 'false'
+  )
+
+  // Methods
+  const handleOpenExternalLink = (url: string) => {
+    if (chrome.tabs !== undefined) {
+      chrome.tabs.create({ url }, () => {
+        if (chrome.runtime.lastError) {
+          console.error(
+            'tabs.create failed: ' + chrome.runtime.lastError.message
+          )
+        }
+      })
+    } else {
+      // Tabs.create is desktop specific. Using window.open for android.
+      window.open(url, '_blank', 'noopener noreferrer')
+    }
+  }
```

Comment:
Instead of adding an extra way to open URLs, shouldn't we just use the existing mojom `api.uiHandler.openURL(url)` which is already security approved (has a scheme check and opens in the approved way)?

Reply:
Will be addressed in https://github.com/brave/brave-core/pull/28620